### ISSUE TITLE
Why test mAP changed? 

### ISSUE THREAD

**Issue by minan19605** [facebookresearch/detectron2#1798]
I  used detectron2 Cascasde-mask-rcnn to train myself dataset and got mAP about 0.75 for BBox and segm at day 5/16/2020.
Today I use the same trained model to run test procedure again but got mAP only about 0.22. The model and dataset don't change.
Could you advise what factors may affect the test result?   In the attached log file, there has full information.
Thanks a lot!
[log.txt](https://github.com/facebookresearch/detectron2/files/4951159/log.txt)

## Expected behavior:

If there are no obvious error in "what you observed" provided above,
please tell us the expected behavior.

If you expect the model to converge / work better, note that we do not give suggestions
on how to train a new model.
Only in one of the two conditions we will help with it:
(1) You're unable to reproduce the results in detectron2 model zoo.
(2) It indicates a detectron2 bug.

## Environment:

Provide your environment information using the following command:
```
wget -nc -q https://github.com/facebookresearch/detectron2/raw/master/detectron2/utils/collect_env.py && python collect_env.py
```

If your issue looks like an installation issue / environment issue,
please first try to solve it with the instructions in
https://detectron2.readthedocs.io/tutorials/install.html#common-installation-issues

**Comment by ppwwyyxx** [comment]
The log seems to be a mix of multiple training & testing runs with even different architectures so it's difficult to make sense of it. If you're still able to reproduce the good AP using the code at 05/16, it sounds like a potential problem in detectron2, you might be able to check (e.g., by bisection) which commit causes this. 

The official model still produces correct result, so if there is indeed an issue, it is likely specific to the dataset. Certain suspicious commits include https://github.com/facebookresearch/detectron2/commit/d4829a673976b83cb949b827bb151d9e77103862 and https://github.com/facebookresearch/detectron2/commit/75ebc5340fa31f9ee0f41e9d6b761db249237043, which you might want to check.

**Comment by minan19605** [comment]
1. Seems when the output directory is the same, detectron2 output log in same log file. That is why the log file look like a multiple train test runs. Actually, I successfully trained a model by using configs/Misc/cascade_mask_rcnn_R_50_FPN_3x.yaml .
2. the first test result in the day 5/16 and is at the line 27054 with the AP values:
[05/16 08:17:54] d2.evaluation.coco_evaluation INFO: Evaluation results for bbox: 
|   AP   |  AP50  |  AP75  |  APs  |  APm   |  APl   |
|:------:|:------:|:------:|:-----:|:------:|:------:|
| 52.361 | 75.466 | 63.058 |  nan  | 11.980 | 52.852 |
[05/16 08:17:54] d2.evaluation.coco_evaluation INFO: Note that some metrics cannot be computed.
[05/16 08:18:01] d2.evaluation.coco_evaluation INFO: Evaluation results for segm: 
|   AP   |  AP50  |  AP75  |  APs  |  APm  |  APl   |
|:------:|:------:|:------:|:-----:|:-----:|:------:|
| 45.259 | 75.781 | 54.230 |  nan  | 3.415 | 45.773 |

3. Then I run the same script at day 7/20 . the script is:
python tools/plain_train_net.py \
	--config-file ./configs/Misc/cascade_mask_rcnn_R_50_FPN_3x.yaml \
	--num-gpus 1 \
	--resume \
	OUTPUT_DIR ./output/cascade/20200516/ SOLVER.IMS_PER_BATCH 2 SOLVER.BASE_LR 0.0025 MODEL.RPN.BATCH_SIZE_PER_IMAGE 1024

Because there already exist the file: model_final.pth. So it just run the test procedure and get AP50 only 22.8.

[07/20 13:02:28] d2.evaluation.coco_evaluation INFO: Evaluation results for bbox: 
|   AP   |  AP50  |  AP75  |  APs  |  APm  |  APl   |
|:------:|:------:|:------:|:-----:|:-----:|:------:|
| 16.280 | 22.885 | 18.881 |  nan  | 0.000 | 16.312 |
[07/20 13:02:28] d2.evaluation.coco_evaluation INFO: Note that some metrics cannot be computed.
[07/20 13:02:30] d2.evaluation.coco_evaluation INFO: Evaluation results for segm: 
|   AP   |  AP50  |  AP75  |  APs  |  APm  |  APl   |
|:------:|:------:|:------:|:-----:|:-----:|:------:|
| 13.793 | 22.892 | 16.713 |  nan  | 0.000 | 13.890 |

At day 7/21, I create a new condo env, install pytorch 1.51 version, and git clone latest detectron2 code. Do test again, but have same result, AP50 is 22.8.

I found one issue, the first test used about 12 minutes, but recent tests, only use 2:30 minutes.

I am using my dataset from the detectron1, it works well.

I modify the plain_train_net.py, the setup() for register myself dataset.

So could you please give some more hints to help me find the issues?  
[plain_train_net.py.txt](https://github.com/facebookresearch/detectron2/files/4957736/plain_train_net.py.txt)

**Comment by ppwwyyxx** [comment]
The first thing to do, as suggested above, is to check if you're still able to reproduce the good AP using the code at 05/16 (or earlier).

**Comment by minan19605** [comment]
I can't reproduce the good AP now.  From 5/16 to 7/20, I didn't change the detectron2, but I did upgrade Nvidia driver version, upgrade Pytorch version from V1.5 to V1.5.1.

**Comment by minan19605** [comment]
I just use the new installed conda env to train the Cascase-mask-r-cnn model again.  That needs about 46 hours.  

The new env includes: Cuda 11.0,  Nvidia driver 450.XX , pytorch 1.5.1 , and latest detectron2 code.

**Comment by ppwwyyxx** [comment]
From the information given it doesn't sound like this is caused by detectron2. 
Perhaps detectron2 or pytorch doesn't work perfectly with cuda 11, or there are other softwares/drivers that don't work well together. If you have more findings that show it's related to detectron2, feel free to share them.
